### PR TITLE
Match some bow funcs, a laki, and a chest func

### PR DESCRIPTION
**Pull request by ThreeMustyFears**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀 This PR matches 7 functions (+0.13%) on `us`.

⚠️ This PR introduces 17 warnings:
- src/entity/Chest.c:153: warning: initialization from incompatible pointer type
- src/entity/Chest.c:169: warning: implicit declaration of function `entity_Chest_open'
- src/entity/Chest.c:229: warning: passing arg 6 of `playFX_1B' makes integer from pointer without a cast
- src/world/partner/bow.c:46: warning: assignment makes pointer from integer without a cast
- src/world/partner/lakilester.c:123: warning: initialization from incompatible pointer type
- src/world/partner/lakilester.c:36: warning: assignment makes pointer from integer without a cast
- src/world/partner/../../npc.h:185: warning: function declaration isn't a prototype
- src/world/partner/../../npc.h:189: warning: function declaration isn't a prototype
- src/world/partner/../../npc.h:210: warning: redundant redeclaration of `npc_dyn_entity_draw_no_op' in same scope
- src/world/partner/../../npc.h:259: warning: redundant redeclaration of `get_enemy' in same scope
- src/world/partner/../../npc.h:35: warning: redundant redeclaration of `get_npc_by_index' in same scope
- src/world/partner/../../npc.h:51: warning: function declaration isn't a prototype
- src/world/partner/../../npc.h:58: warning: redundant redeclaration of `npc_move_heading' in same scope
- src/world/partner/../../npc.h:60: warning: redundant redeclaration of `get_npc_unsafe' in same scope
- src/world/partner/../../npc.h:63: warning: redundant redeclaration of `get_npc_safe' in same scope
- src/world/partner/../../npc.h:81: warning: redundant redeclaration of `set_npc_yaw' in same scope

File: src/entity/Chest.c [hunk 1]
```diff
@@ -145,7 +148,97 @@ void entity_Chest_close(Entity* entity) {
...
+                }
+                chest->unk_1C -= temp_f4;
+            }
+            delta = sin_rad(chest->unk_24 * TAU / 360.0f);
+            theta = intermediateTheta = clamp_angle(atan2(entity->position.x, entity->position.z,
+                                                          playerStatus->position.x, playerStatus->position.z));
+
+            if (gGameStatusPtr->areaID == 0x12) {
+                phi_f22 = 3.0f;
+            } else phi_f22 = 4.0f;
```
Comment (ethteck) [inline src/entity/Chest.c:205]: curly braces around this else please

File: src/world/partner/bow.c [hunk 2]
```diff
@@ -97,14 +105,246 @@ s32 func_802BD168_323AB8(Evt* evt, s32 arg1) {
...
+        partner->flags = D_802BDFFC_32494C->unk_08;
+        D_802BDFFC_32494C->unk_04 = 0;
+        partner_clear_player_tracking(partner);
+    }
+}
 
 INCLUDE_ASM(s32, "world/partner/bow", func_802BD540_323E90);
 
-INCLUDE_ASM(s32, "world/partner/bow", func_802BD694_323FE4);
+s32 func_802BD694_323FE4(Evt* arg0, s32 arg1) {
```
Comment (ethteck) [inline src/world/partner/bow.c:119]: arg0 = script, arg1 = isInitialCall

File: src/world/partner/bow.c [hunk 3]
```diff
@@ -97,14 +105,246 @@ s32 func_802BD168_323AB8(Evt* evt, s32 arg1) {
...
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    PartnerActionStatus* partnerActionStatus = &gPartnerActionStatus;
+    CollisionStatus* collisionStatus = &gCollisionStatus;
+    Npc* npc;
+    f32 distance;
+
+    npc = arg0->owner2.npc;
+    if (arg1 != 0) {
```
Comment (ethteck) [inline src/world/partner/bow.c:127]: != 0 isn't needed for this bool check

File: src/world/partner/bow.c [hunk 4]
```diff
@@ -97,14 +105,246 @@ s32 func_802BD168_323AB8(Evt* evt, s32 arg1) {
...
 
-INCLUDE_ASM(s32, "world/partner/bow", func_802BD694_323FE4);
+s32 func_802BD694_323FE4(Evt* arg0, s32 arg1) {
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    PartnerActionStatus* partnerActionStatus = &gPartnerActionStatus;
+    CollisionStatus* collisionStatus = &gCollisionStatus;
+    Npc* npc;
+    f32 distance;
+
+    npc = arg0->owner2.npc;
```
Comment (ethteck) [inline src/world/partner/bow.c:126]: npc can be initialized on the line where it's declared

File: src/world/partner/bow.c [hunk 5]
```diff
@@ -97,14 +105,246 @@ s32 func_802BD168_323AB8(Evt* evt, s32 arg1) {
...
+                if (playerStatus->animFlags & 0x200000) {
+                    playerStatus->animFlags &= ~0x200000;
+                    arg0->functionTemp[2] = disable_player_input();
+                    D_802BE0C4 = 1;
+                    arg0->functionTemp[0] = 20;
+                } else {
+                    arg0->functionTemp[0] = 40;
+                }
+            } else {
+                return 2;
```
Comment (ethteck) [inline src/world/partner/bow.c:140]: ApiStatus_DONE2 (below as well)

File: src/world/partner/bow.c [hunk 6]
```diff
@@ -97,14 +105,246 @@ s32 func_802BD168_323AB8(Evt* evt, s32 arg1) {
...
+                    arg0->functionTemp[0] = 1;
+                }
+                break;
+            }
+            func_802BDDF0_324740(npc);
+            return 2;
+        case 1:
+            if (collisionStatus->currentFloor >= 0) {
+                playerStatus->alpha1 -= 8;
+                if (playerStatus->alpha1 < 0x81) {
```
Comment (ethteck) [inline src/world/partner/bow.c:243]: <= 0x80

File: src/world/partner/bow.c [hunk 7]
```diff
@@ -97,14 +105,246 @@ s32 func_802BD168_323AB8(Evt* evt, s32 arg1) {
...
+                    return 2;
+                }
+                arg0->functionTemp[0] = 20;
+            }
+            break;
+    }
+
+    switch (arg0->functionTemp[0]) {
+        case 20:
+            if ((playerStatus->flags & 0x800) != 0) {
```
Comment (ethteck) [inline src/world/partner/bow.c:185]: != 0 and parens not needed

File: src/world/partner/bow.c [hunk 8]
```diff
@@ -97,14 +105,246 @@ s32 func_802BD168_323AB8(Evt* evt, s32 arg1) {
...
+    }
+    return 0;
+}
+
+void func_802BDDF0_324740(Npc* partner) {
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    PartnerActionStatus* partnerActionStatus = &gPartnerActionStatus;
+    s32 actionState;
+
+    playerStatus->alpha1 = 0xFF;
```
Comment (ethteck) [inline src/world/partner/bow.c:301]: alphas can be in decimal (255)

File: src/world/partner/bow.c [hunk 9]
```diff
@@ -97,14 +105,246 @@ s32 func_802BD168_323AB8(Evt* evt, s32 arg1) {
...
+    if (D_802BE0C4 != 0) {
+        enable_player_input();
+    }
 
-INCLUDE_ASM(s32, "world/partner/bow", func_802BDF08_324858);
+    playerStatus->flags &= ~0x8002;
+    partner->flags &= ~0x42;
+    D_802BE0C4 = 0;
+    actionState = ACTION_STATE_IDLE;
+    if ((playerStatus->flags & 0x800) != 0) {
```
Comment (ethteck) [inline src/world/partner/bow.c:314]: != 0 and parens can be removed

File: src/world/partner/bow.c [hunk 10]
```diff
@@ -97,14 +105,246 @@ s32 func_802BD168_323AB8(Evt* evt, s32 arg1) {
...
-INCLUDE_ASM(s32, "world/partner/bow", func_802BDF64_3248B4);
+    set_action_state(actionState);
+    partnerActionStatus->actionState.b[0] = 0;
+    partnerActionStatus->actionState.b[3] = 0;
+    playerStatus->flags &= ~0x100;
+    partner_clear_player_tracking(partner);
+    D_802BE0C0 = 0;
+}
+
+s32 func_802BDF08_324858(Evt* arg0, s32 arg1) {
```
Comment (ethteck) [inline src/world/partner/bow.c:326]: standard script function signature (return ApiStatus, args, etc)

File: src/world/partner/bow.c [hunk 11]
```diff
@@ -97,14 +105,246 @@ s32 func_802BD168_323AB8(Evt* evt, s32 arg1) {
...
+    Npc* partner = arg0->owner2.npc;
+    if (arg1 != 0) {
```
Comment (ethteck) [inline src/world/partner/bow.c:328]: Empty line after this, no need for != 0 for bool

File: src/world/partner/lakilester.c [hunk 12]
```diff
@@ -109,7 +117,138 @@ INCLUDE_ASM(s32, "world/partner/lakilester", func_802BE6A0_3221F0);
 
 INCLUDE_ASM(s32, "world/partner/lakilester", func_802BE724_322274);
 
-INCLUDE_ASM(s32, "world/partner/lakilester", func_802BF4F0_323040);
+s32 func_802BF4F0_323040(Evt* arg0, s32 arg1) {
```
Comment (ethteck) [inline src/world/partner/lakilester.c:120]: Script func

**Review by ethteck — CHANGES_REQUESTED** [review]

File: src/world/partner/bow.c [hunk 13]
```diff
@@ -97,14 +104,246 @@ s32 func_802BD168_323AB8(Evt* evt, s32 arg1) {
...
 
-INCLUDE_ASM(s32, "world/partner/bow", func_802BDF64_3248B4);
+    if (isInitialCall) {
+        partner_init_put_away(partner);
+        if (D_802BE0C0 != 0) {
+            sfx_play_sound_at_npc(SOUND_BOW_APPEAR, 0, -4);
+        }
+        func_802BDDF0_324740(partner);
+    }
+    return partner_put_away(partner) != 0;
```
Comment (nanaian) [inline src/world/partner/bow.c:334]:
```suggestion
    if (partner_put_away(partner)) {
        return ApiStatus_DONE1;
    } else {
        return ApiStatus_BLOCK;
    }
```

File: src/world/partner/bow.c [hunk 14]
```diff
@@ -97,14 +104,246 @@ s32 func_802BD168_323AB8(Evt* evt, s32 arg1) {
...
+}
+
+void func_802BDF64_3248B4(Npc* partner) {
+    PartnerActionStatus* partnerActionStatus = &gPartnerActionStatus;
+
+    if (D_802BE0C0 != 0) {
```
Comment (nanaian) [inline src/world/partner/bow.c:340]: This seems to be a bool?
```suggestion
    if (D_802BE0C0) {
```

File: src/world/partner/bow.c [hunk 15]
```diff
@@ -97,14 +104,246 @@ s32 func_802BD168_323AB8(Evt* evt, s32 arg1) {
...
+        enable_player_input();
+        set_action_state(ACTION_STATE_IDLE);
+        partner_clear_player_tracking(partner);
+        partnerActionStatus->actionState.b[0] = 0;
+        partnerActionStatus->actionState.b[3] = 0;
+        D_802BE0C0 = 0;
```
Comment (nanaian) [inline src/world/partner/bow.c:346]:
```suggestion
        D_802BE0C0 = FALSE;
```

File: src/world/partner/lakilester.c [hunk 16]
```diff
@@ -109,7 +116,138 @@ INCLUDE_ASM(s32, "world/partner/lakilester", func_802BE6A0_3221F0);
...
+    f32 sp20;
+    f32 yaw;
+    s32 phi_v1;
+
+    if (isInitialCall) {
+        phi_v1 = 0;
+        if (D_802BFF0C == 0) {
+            phi_v1 = 3;
+        }
+        D_802BFF00 = phi_v1;
```
Comment (nanaian) [inline src/world/partner/lakilester.c:136]: If it matches,
```suggestion
        if (D_802BFF0C == 0) {
            D_802BFF00 = 3;
        } else {
            D_802BFF00 = 0;
        }
```

File: src/world/partner/lakilester.c [hunk 17]
```diff
@@ -109,7 +116,138 @@ INCLUDE_ASM(s32, "world/partner/lakilester", func_802BE6A0_3221F0);
...
+
+    gCameras->targetPos.x = playerStatus->position.x;
+    gCameras->targetPos.y = playerStatus->position.y;
+    gCameras->targetPos.z = playerStatus->position.z;
+
+    switch (D_802BFF00) {
+        case 3:
+            partner->flags &= ~0x48;
+            if (D_802BFF08 != 0) {
+                D_802BFF08 = 0;
```
Comment (nanaian) [inline src/world/partner/lakilester.c:205]: Bool?

**Review by ethteck — APPROVED** [review]

File: src/world/partner/lakilester.c [hunk 18]
```diff
@@ -129,9 +129,10 @@ ApiStatus func_802BF4F0_323040(Evt* script, s32 isInitialCall) {
     s32 phi_v1;
 
     if (isInitialCall) {
-        phi_v1 = 0;
         if (D_802BFF0C == 0) {
             phi_v1 = 3;
+        } else {
+            phi_v1 = 0;
         }
         D_802BFF00 = phi_v1;
```
Comment (nanaian) [inline src/world/partner/lakilester.c:137]: Oh does a ternary work?

Comment (ThreeMustyFears) [reply src/world/partner/lakilester.c]: Yeah it seems to

**Pull request merged**